wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -245,7 +293,8 @@ private[wallet] trait TransactionProcessing extends WalletLogger {
     */
   private def processNewIncomingTx(
       transaction: Transaction,
-      confirmations: Int): Future[Seq[SpendingInfoDb]] = {
+      confirmations: Int,
+      blockHashOpt: Option[DoubleSha256DigestBE]): Future[Seq[SpendingInfoDb]] = {
```

Comment:
So if `confirmations = 0`, `blockHashOpt = None` right? Maybe we should be making a helper case class like this:

```scala
/** This represents a block hash combined with the number of confirmations
  * a transaction should have relative to this block hash
  * 
  * I.e. 
  * If a transaction was included 10 blocks ago, this case class would be 
  * {{{
  *   BlockHashWithConfirmations(hash, 10)
  * }}}
  * 
  * */
case class BlockHashWithConfirmations(blockHash: DoubleSha256DigestBE, confirmations: Int)
```

At least I think that is what is intended here?

Reply:
I'm going to remove `confirmations` soon. I still have them to let some of the tests pass.